rest/utilities_testing.go

```diff
@@ -721,6 +807,152 @@ func (bt *BlipTester) GetChanges() (changes [][]interface{}) {
 
 }
 
+
+func (bt *BlipTester) WaitForNumDocsViaChanges(numDocsExpected int) (docs map[string]RestDocument) {
+
+	retryWorker := func() (shouldRetry bool, err error, value interface{}) {
+			allDocs := bt.GetAllDocsViaChanges()
+			if len(allDocs) >= numDocsExpected {
+				return false, nil, allDocs
+			}
+
+			// haven't seen numDocsExpected yet, so wait and retry
+			return true, nil, nil
+
+	}
+
+	_, allDocs := base.RetryLoop(
+		"WaitForNumDocsViaChanges",
+		retryWorker,
+		base.CreateDoublingSleeperFunc(10, 10),
+		)
+
+	return allDocs.(map[string]RestDocument)
+
+}
+
+// Get all documents and their attachments via the following steps:
+//
+// - Invoking one-shot subChanges request
+// - Responding to all incoming "changes" requests from peer to request the changed rev, and accumulate rev body
+// - Responding to all incoming "rev" requests from peer to get all attachments, and accumulate them
+// - Return accumulated docs + attachements to caller
```

Comment:
This sounds like it's a one-shot pull replication (w/out checkpoints)?  If so, can we rename it to something that reflects that?  GetAllDocsViaChanges() makes me think of either changes w/ include_docs, or an _all_docs request.

Reply:
Having said that - if this IS just a one-shot pull replication, should it be defined as part of a blip_sync client implementation, instead of as a test helper function?

Reply:
Yeah, it's essentially a pull replication w/o the checkpointing, although it just pulls the latest revision of each doc.  I'll go ahead and rename to `PullDocs()` since I agree `GetAllDocsViaChanges()` is confusing/misleading.  

In terms of defining in a blip_sync client implementation, I agree, but there's not currently a place for that code that I know of.  The closes thing I can think of would be: https://github.com/couchbaselabs/sg-blip  

Were you thinking just sticking this in separate SG package, or a full separate repo?

Reply:
I was just thinking of a separate SG file, but structure its API to look more like a client replication API.